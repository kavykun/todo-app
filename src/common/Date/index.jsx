import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './Date.css';

const Date = (props) => {
  const { date } = props;

  if (!date || date === '') {
    return null;
  }

  const dateFormat = new window.Date(date);

  const formattedDate = moment(dateFormat).utc().format('MM/DD/YYYY');

  return <div className="date">{formattedDate}</div>;
};

Date.defaultProps = {
  date: null,
};

Date.propTypes = {
  date: PropTypes.string,
};

export default Date;
