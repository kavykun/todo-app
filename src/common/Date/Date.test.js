import { render } from '@testing-library/react';

import Date from './index';

test('Should return an empty div if date is empty string', async () => {
  const date = '';
  const { asFragment } = render(<Date date={date} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Should return an empty div if date is null', async () => {
  const date = null;
  const { asFragment } = render(<Date date={date} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Should match snapshot', async () => {
  const date = '07/05/2012';
  const { asFragment } = render(<Date date={date} />);
  expect(asFragment()).toMatchSnapshot();
});
