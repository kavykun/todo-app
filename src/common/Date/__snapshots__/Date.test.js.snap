// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="date"
  >
    07/05/2012
  </div>
</DocumentFragment>
`;

exports[`Should return an empty div if date is empty string 1`] = `<DocumentFragment />`;

exports[`Should return an empty div if date is null 1`] = `<DocumentFragment />`;
